jetty-webapp/src/main/java/org/eclipse/jetty/webapp/ClasspathPattern.java

```diff
@@ -685,21 +686,7 @@ public boolean match(Class<?> clazz)
     {       
         try
         {
-            Boolean byName = _patterns.isIncludedAndNotExcluded(clazz.getName());
-            if (Boolean.FALSE.equals(byName))
-                return byName; // Already excluded so no need to check location.
-            URI location = TypeUtil.getLocationOfClass(clazz);
-            Boolean byLocation = location == null ? null
-                    : _locations.isIncludedAndNotExcluded(location);
-            
-            if (LOG.isDebugEnabled())
-                LOG.debug("match {} from {} byName={} byLocation={} in {}",clazz,location,byName,byLocation,this);
-            
-            // Combine the tri-state match of both IncludeExclude Sets
-            boolean included = Boolean.TRUE.equals(byName) || Boolean.TRUE.equals(byLocation)
-                || (byName==null && !_patterns.hasIncludes() && byLocation==null && !_locations.hasIncludes());
-            boolean excluded = Boolean.FALSE.equals(byName) || Boolean.FALSE.equals(byLocation);
-            return included && !excluded;
+            return IncludeExcludeSet.or(_packageOrNamePatterns, clazz.getName(), _locations, ()->TypeUtil.getLocationOfClass(clazz));
```

Comment:
Too difficult to understand, see above.

Reply:
Consider the example of pattern 'exclude com.acme.*' and location 'include /foo/bar.jar'.
The 'com.example.MyClass' loaded from '/some/other.jar'  would pass a simple or operation, but it must be excluded because it is not loaded from '/foo/bar.jar'.  It is not a simple OR operation.   But perhaps it is a simple AND operation elsewhere...